src/main/java/seedu/address/model/group/NameContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.group;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class NameContainsKeywordsPredicate implements Predicate<Group> {
```

Comment:
This class may overlap with Ye Ji's class in functionality. I don't think it will cause merge conflicts, but it would be best not to have multiple classes doing the same thing.